### DISCUSSION TITLE
Support for VolumeSource images in future

### DISCUSSION THREAD

**Discussion by danielloader** [kubernetes-sigs/kustomize#5750] [Q&A]
Thought I'd ask early and see, it might even work already!

[KEP 4639](https://github.com/kubernetes/enhancements/issues/4639) recently got pushed into alpha with a working implementation in 1.31 (behind the alpha feature gate).

Just wondering if the existing logic for matching images and tags would work with the object schema found in the [docs](https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/)?

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: image-volume
spec:
  containers:
  - name: shell
    command: ["sleep", "infinity"]
    image: debian
    volumeMounts:
    - name: volume
      mountPath: /volume
  volumes:
  - name: volume
    image:
      reference: quay.io/crio/artifact:v1
      pullPolicy: IfNotPresent
```

To summarise, I'd love to be able to patch the `.spec.volumes[0].image.reference` field with the existing image functionality in kustomize.
